DQMOffline/Trigger/plugins/NoBPTXMonitor.cc

```diff
@@ -0,0 +1,401 @@
+  for ( auto const & j : *jetHandle ) {
+    if ( jetSelection_( j ) ) jets.push_back(j);
+  }
+  if ((unsigned int)(jets.size()) < njets_ ) return;
+  double jetE = -999;
+  double jetEta = -999;
+  double jetPhi = -999;
+  if(jets.size()>0){
+    jetE = jets[0].energy();
+    jetEta = jets[0].eta();
+    jetPhi = jets[0].phi();
+  }
+
+  edm::Handle<reco::TrackCollection> DSAHandle;
+  iEvent.getByToken( muonToken_, DSAHandle );
+  if ((unsigned int)(DSAHandle->size()) < nmuons_ ) return;
+  std::vector<reco::Track> muons;
+  for ( auto const & m : *DSAHandle ) {
+    if ( muonSelection_( m ) ) muons.push_back(m);
+  }
+  if ((unsigned int)(muons.size()) < nmuons_ ) return;
+  double muonPt = -999;
+  double muonEta = -999;
+  double muonPhi = -999;
+  if(muons.size()>0){
+    muonPt = muons[0].pt();
+    muonPt = muons[0].eta();
+    muonPt = muons[0].phi();
+  }
+
+  // filling histograms (denominator)  
+  jetENoBPTX_.denominator -> Fill(jetE);
+  jetENoBPTX_variableBinning_.denominator -> Fill(jetE);
+  jetEVsLS_.denominator -> Fill(ls, jetE);
+  jetEVsBX_.denominator -> Fill(bx, jetE);
+  jetEtaNoBPTX_.denominator -> Fill(jetEta);
+  jetEtaVsLS_.denominator -> Fill(ls, jetEta);
+  jetEtaVsBX_.denominator -> Fill(bx, jetEta);
+  jetPhiNoBPTX_.denominator -> Fill(jetPhi);
+  jetPhiVsLS_.denominator -> Fill(ls, jetPhi);
+  jetPhiVsBX_.denominator -> Fill(bx, jetPhi);
+  muonPtNoBPTX_.denominator -> Fill(muonPt);
+  muonPtNoBPTX_variableBinning_.denominator -> Fill(muonPt);
+  muonPtVsLS_.denominator -> Fill(ls, muonPt);
+  muonPtVsBX_.denominator -> Fill(bx, muonPt);
+  muonEtaNoBPTX_.denominator -> Fill(muonEta);
+  muonEtaVsLS_.denominator -> Fill(ls, muonEta);
+  muonEtaVsBX_.denominator -> Fill(bx, muonEta);
+  muonPhiNoBPTX_.denominator -> Fill(muonPhi);
+  muonPhiVsLS_.denominator -> Fill(ls, muonPhi);
+  muonPhiVsBX_.denominator -> Fill(bx, muonPhi);
+
+  // filling histograms (numerator)  
+  if (num_genTriggerEventFlag_->on() && ! num_genTriggerEventFlag_->accept( iEvent, iSetup) ) return;
+  jetENoBPTX_.numerator -> Fill(jetE);
+  jetENoBPTX_variableBinning_.numerator -> Fill(jetE);
+  jetEVsLS_.numerator -> Fill(ls, jetE);
+  jetEVsBX_.numerator -> Fill(bx, jetE);
+  jetEtaNoBPTX_.numerator -> Fill(jetEta);
+  jetEtaVsLS_.numerator -> Fill(ls, jetEta);
+  jetEtaVsBX_.numerator -> Fill(bx, jetEta);
+  jetPhiNoBPTX_.numerator -> Fill(jetPhi);
+  jetPhiVsLS_.numerator -> Fill(ls, jetPhi);
+  jetPhiVsBX_.numerator -> Fill(bx, jetPhi);
+  muonPtNoBPTX_.numerator -> Fill(muonPt);
+  muonPtNoBPTX_variableBinning_.numerator -> Fill(muonPt);
+  muonPtVsLS_.numerator -> Fill(ls, muonPt);
+  muonPtVsBX_.numerator -> Fill(bx, muonPt);
+  muonEtaNoBPTX_.numerator -> Fill(muonEta);
+  muonEtaVsLS_.numerator -> Fill(ls, muonEta);
+  muonEtaVsBX_.numerator -> Fill(bx, muonEta);
+  muonPhiNoBPTX_.numerator -> Fill(muonPhi);
+  muonPhiVsLS_.numerator -> Fill(ls, muonPhi);
+  muonPhiVsBX_.numerator -> Fill(bx, muonPhi);
+
+}
+
```

Comment:
Well I guess if we are just interested in the efficiency vs BX or LS, then you are right. I guess I was thinking it might be useful to see the jet E, eta, and phi vs BX and LS...